src/pip/_internal/resolution/resolvelib/candidates.py

```diff
@@ -405,8 +405,7 @@ def get_dependencies(self):
         ]
         # Add a dependency on the exact base.
         # (See note 2b in the class docstring)
-        spec = "{}=={}".format(self.base.name, self.base.version)
-        deps.append(factory.make_requirement_from_spec(spec, self.base._ireq))
+        deps.append(factory.make_requirement_from_install_req(self.base._ireq))
```

Comment:
If the base is foo 1.0, derived from a requirement of "foo", won't `self.base._ireq` be a specifier of "foo", whereas we want to be explicit that we depend on foo 1.0 (to avoid ending up trying to install foo 1.0 and foo[bar] 2.0)?

I think we should have a test for this, just to make sure we don't introduce a bug here, but I'm not entirely sure how to write one :-(

Reply:
Hmm, you’re right. We’ll need to have a different implementation to handle this correctly.

Reply:
I decided to remove this patch for now (and add a FIXME comment). I’m suspecting maybe we can simply do `ExplicitRequirement(self.base)` here, but that is worth its own tracking issue.